### PR TITLE
Only allow streaming image assets with supported pixel formats to be selected for an image gradient

### PR DESCRIPTION
**Pull request by cgalvan**
Our Image Gradient component now uses a `AZ::RPI::StreamingImageAsset` for its image asset. But, only a subset of the total pixel formats for `AZ::RPI::StreamingImageAsset` are supported by the pixel retrieval API that the Image Gradient uses. So, the Image Gradient component now uses a custom asset property controller that behaves like the default asset property controller, with the exception that it prevents the user from selecting an asset that has a pixel format that isn't supported by the pixel retrieval API. To do this, the "OK" button on the picker is disabled for incompatible formats.

ImageGradientCustomAssetPicker

Tested and verified that other streaming image assets still use the default asset property controller. Also fixed a logic bug in the default asset picker where the "OK" button was always disabled when first launched, even if it was launched with a pre-existing valid selection.

Signed-off-by: Chris Galvan <chgalvan@amazon.com>

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetPicker/AssetPickerDialog.cpp [hunk 1]
```diff
@@ -174,8 +174,15 @@ namespace AzToolsFramework
                 m_tableModel->UpdateTableModelMaps();
             }
 
-            QTimer::singleShot(0, this, &AssetPickerDialog::RestoreState);
-            SelectionChangedSlot();
+            QTimer::singleShot(0, this, [this]() {
+                RestoreState();
+
+                // The selection doesn't propogate immediately, so we need to delay
```
Comment (mbalfour-amzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetPicker/AssetPickerDialog.cpp:180]: Nit: "propogate" -> "propagate"

**Review by VickyAtAZ — COMMENTED** [review]
The change with RPIUtils files look fine.

**Review by mbalfour-amzn — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetPicker/AssetPickerDialog.cpp (hunk 1 above)
Comment (cgalvan) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetPicker/AssetPickerDialog.cpp]: Argh, thanks! Fixed

**Review by AMZN-daimini — APPROVED** [review]
Looks ok code-wise.

Not necessarily a blocker, but I'd appreciate if the Dialog would show a message stating "Asset type not supported" or another message to that effect somewhere instead of just disabling the Ok button. 

Also make sure pressing Enter or other shortcuts don't still let you select an incorrect value.

**Comment by cgalvan** [comment]
> Looks ok code-wise.
> 
> Not necessarily a blocker, but I'd appreciate if the Dialog would show a message stating "Asset type not supported" or another message to that effect somewhere instead of just disabling the Ok button.
> 
> Also make sure pressing Enter or other shortcuts don't still let you select an incorrect value.

Yeah there is a follow-up task to actually to make use of the Texture Settings Editor to give the user the option to convert the image to a supported format if they try to select one that isn't supported. So instead of the disabled "OK" button, they would we'd replace it with "Edit" or "Convert".

And yep pressing Enter, and double click as well all go through the same `EvaluateSelection` check so they are blocked as well.

**Pull request merged**